### PR TITLE
improve perfparser performance

### PR DESCRIPTION
**Pull request by lievenhey**
these filter only need to be called ones and not in an loop

### CODE REVIEW

File: src/parsers/perf/perfparser.cpp [hunk 1]
```diff
@@ -1604,33 +1608,50 @@ void PerfParser::filterResults(const Data::FilterAction& filter)
...
-                                                           return false;
-                                                       }
-                                                       includedBinaries.remove(symbol.binary);
-                                                       // only stop when we included everything and no exclude filter is
-                                                       // set
-                                                       return !includedSymbols.isEmpty() || !filter.excludeSymbols.isEmpty() || includedBinaries.isEmpty() || !filter.excludeBinaries.isEmpty();
-                                                   });
-                    filterStacks[stackId] = !excluded && includedSymbols.isEmpty() && includedBinaries.isEmpty();
+
+                const auto threadCount = QThread::idealThreadCount();
```
Comment (milianw) [inline src/parsers/perf/perfparser.cpp:1612]: this shoudl use `queue.maximumNumberOfThreads()` (assuming that's available?)

File: src/parsers/perf/perfparser.cpp [hunk 2]
```diff
@@ -1570,6 +1571,9 @@ void PerfParser::filterResults(const Data::FilterAction& filter)
     emit parsingStarted();
     using namespace ThreadWeaver;
     stream() << make_job([this, filter]() {
+        Queue queue(this);
+        queue.setMaximumNumberOfThreads(QThread::idealThreadCount());
```
[the remark this answers is not in the document]
Comment (lievenhey) [reply src/parsers/perf/perfparser.cpp:1575]: I found nothing in the docs.
In my case `maximumNumberOfThreads` returns 16 but I only have 8 cores (with hyperthreading)

[the remark this answers is not in the document]
Comment (milianw) [reply src/parsers/perf/perfparser.cpp:1575]: odd, but ok - let's keep it explicit then

**Review by milianw — APPROVED** [review]
ok, lgtm - thanks

**Pull request merged**